src/query/graphite/storage/m3_wrapper.go

```diff
@@ -94,6 +96,32 @@ func NewM3WrappedStorage(
 func TranslateQueryToMatchersWithTerminator(
 	query string,
 ) (models.Matchers, error) {
+	if strings.Contains(query, "**") {
+		// Need to regexp on the entire ID since ** matches over different
+		// graphite path dimensions.
+		globOpts := graphite.GlobOptions{
+			AllowMatchAll: true,
+		}
+		idRegexp, _, err := graphite.ExtendedGlobToRegexPattern(query, globOpts)
+		if err != nil {
+			return nil, err
+		}
+		// Also make sure it's a graphite metric by making sure it has a
+		// __g0__ tag:
+		hasFirstPathMatcher, err := convertMetricPartToMatcher(0, wildcard)
+		if err != nil {
+			return nil, err
+		}
+		return models.Matchers{
+			hasFirstPathMatcher,
+			models.Matcher{
+				Type:  models.MatchRegexp,
+				Name:  doc.IDReservedFieldName,
```

Comment:
Will looking in the ID field for this fail for search patterns with a "." in it?

Reply:
`ExtendedGlobToRegexPattern` which we call here takes care of this:
```go
		case '.':
			// Match hierarchy separator
			pattern.WriteString("\\.+")
			regexed = true
```

i.e. it escapes the dot and allows for searching patterns with dots in it.